authors/apps/profiles/views.py

```diff
@@ -0,0 +1,35 @@
+from rest_framework import generics, status
+from rest_framework.response import Response
+from django.core.exceptions import ObjectDoesNotExist
+from .serializers import UserProfileSerializer
+from .models import UserProfile
+
+
+class ProfileView(generics.RetrieveUpdateAPIView):
+    '''
+    class view to update user profile
+    '''
+    serializer_class = UserProfileSerializer
+    queryset = UserProfile.objects.all()
+
+    def update(self, request, username, *args, **kwargs):
+        try:
+            user_instance = UserProfile.objects.select_related(
+                'user').get(user__username=username)
+            if user_instance.user.username != request.user.username:
+                return Response({'error':'You are not allowed to edit or delete this object'},status.HTTP_403_FORBIDDEN)
+        except ObjectDoesNotExist:
+            return Response({'error': 'Username {} not found'.format(username)}, status.HTTP_404_NOT_FOUND)
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.update(instance=user_instance, validated_data=request.data)
+        return Response({'message': 'Profile has been updated successfully'},status.HTTP_200_OK)
```

Comment:
Please return the user's profile after an update, so that on the frontend we won't need need to make a get request to get the new user details